### PR TITLE
Feature google login

### PR DESCRIPTION
**Pull request by vijivasanthan**

### CODE REVIEW

**Comment by vijivasanthan** [comment]
[(8) 8B3: Allow usage of Google Identity Provider to login](https://trello.com/c/uOSrm4oA/1031-8-8b3-allow-usage-of-google-identity-provider-to-login)

File: app/layout.jade [hunk 1]
```diff
@@ -30,6 +30,19 @@ html.no-js(lang='en')
 			}
 		script(src="https://maps.google.com/maps/api/js")
 		script(src="scripts/localConfig.js")
+		script(src="https://apis.google.com/js/platform.js" async defer)
+		script(src="https://www.gstatic.com/firebasejs/4.9.0/firebase-app.js")
+		script(src="https://www.gstatic.com/firebasejs/4.9.0/firebase-auth.js")
+		script(src="https://www.gstatic.com/firebasejs/4.9.0/firebase-database.js")
```
Comment (leonjoosse) [inline app/layout.jade]: Are you sure you need the 'firebase-database' dependency?

File: app/scripts/services/login.js [hunk 2]
```diff
@@ -256,48 +259,138 @@ define(['services/services', 'config'],
...
+                              };
+
+                              /**
+                               * Link GoogleAccount with TeamTelefoon
+                               */
+                              this.linkGoogleAccount = function (loginInfo) {
+                                var deferred = $q.defer();
+                                var self = this;
+                                var result;
+                                var linkGoogleTT = $resource('https://test-backend.teamtelefoon.nl/' + 'authenticate/google', {}, {
```
Comment (leonjoosse) [inline app/scripts/services/login.js]: Why is the test-backend hardcoded here?

File: app/layout.jade [hunk 3]
```diff
@@ -30,6 +30,19 @@ html.no-js(lang='en')
...
+		script(src="https://www.gstatic.com/firebasejs/4.9.0/firebase.js")
+		script.
+			// Initialize Firebase
```
Comment (leonjoosse) [inline app/layout.jade]: This must be stored in the localconfig file, not publicly on Github. 
After storing it in the localconfig file, you can still retrieve it when visiting the web page, but it is less searchable than on Github.

File: app/scripts/controllers/login/login.js [hunk 4]
```diff
@@ -58,30 +96,82 @@ define(
...
           }
         };
 
         /**
-         * @param errorMessage The message who will show up
-         * @param display If it's displayed or not
-         * @returns {{login: {display: boolean, message: (string|string|string|string)}}}
+         * initApp() called onload for signing in with google.
+         * firebase.auth().getRedirectResult(result) gets back the user details after signing in google application
+         * @returns {{result: {user, credential, operationtype, additionalUserInfo}}}
```
Comment (leonjoosse) [inline app/scripts/controllers/login/login.js]: It is a bit nitpicky, but the function actually doesn't return what the `@return` documentation says. Probably one of the firebase() calls returns that object.

File: app/layout.jade (hunk 1 above)
Comment (vijivasanthan) [reply app/layout.jade]: It's not needed I removed.

File: app/scripts/services/login.js (hunk 2 above)
Comment (vijivasanthan) [reply app/scripts/services/login.js]: By mistake.Replaced

**Pull request merged**